File: lib/services/localization_service.js
const Services = require('./service');
const { Language } = require('./../models/language');
const { DisplayTagName } = require('./../models/displaytagname');

/**
 * Localization Service, Covers all api endpoints related to extracting information
 * about the languages available and displayTagNames
 */
class LocalizationService extends Services {
  constructor(endPoint) {
    super(endPoint);
  }

  async languages(authTokenKey, tenant) {
    let result = await this.executeOperation("/local/languages", {
      authTokenKey: authTokenKey,
      tenant: tenant
    });

    if (result) {
      let entries = result.content.contentList || [];
      return entries.map(entry => {
        return new Language(entry);
      });
    } else {
      throw Error(`failed to list languages of tenant ${tenant}`);
    }
  }  

  async displayTagNames(lastUpdated = "", clientInfo = "") {
    let result = await this.executeOperation("/local/displayTagNames", {
      lastUpdated: lastUpdated,
      clientInfo: clientInfo
    });

    if (result) {
      let entries = result.content.contentList || [];
      return entries.map(entry => {
        return new DisplayTagName(entry);
      });
    } else {
      throw Error(`failed to list users of tenant ${tenant}`);
    }
  }  
}

module.exports = LocalizationService;

File: libraries/java/src/main/java/com/qmplus/v3/api/services/LocalizationService.java
package com.qmplus.v3.api.services;

import com.fasterxml.jackson.core.type.TypeReference;
import com.qmplus.v3.api.models.request.LanguageRequest;
import com.qmplus.v3.api.models.request.LocalizedRequest;
import com.qmplus.v3.api.models.response.*;

import java.io.IOException;

public class LocalizationService extends BaseService {
  LocalizationService() {
    super();
  }

  LocalizationService(String endPoint) {
    super(endPoint);
  }

  public ResponseWrapperList<LanguageResponse> languages(
      String authTokenKey, String tenant, String clientInfo
  ) throws IOException {
    LanguageRequest request = new LanguageRequest();

    // Trigger the default random key
    request.setAuthTokenKey(authTokenKey);
    request.setTenant(tenant);
    request.setClientInfo(clientInfo != null ? clientInfo : "");

    // Execute the operation
    ResponseWrapper<ResponseWrapperList<LanguageResponse>> response
        = executeOperation(new TypeReference<ResponseWrapper<ResponseWrapperList<LanguageResponse>>>() {},"/local/languages", toJson(request));

    // Return the content object if everything went well
    return response.getContent();
  }

  public ResponseWrapperList<LocalizedResponse> displayTagNames(
      String lastUpdated, String clientInfo
  ) throws IOException {
    LocalizedRequest request = new LocalizedRequest();

    // Trigger the default random key
    request.setLastUpdated(lastUpdated != null ? lastUpdated : "");
    request.setClientInfo(clientInfo != null ? clientInfo : "");

    // Execute the operation
    ResponseWrapper<ResponseWrapperList<LocalizedResponse>> response
        = executeOperation(new TypeReference<ResponseWrapper<ResponseWrapperList<LocalizedResponse>>>() {},"/local/displayTagNames", toJson(request));

    // Return the content object if everything went well
    return response.getContent();
  }
}
